src/avfoundation.cs

```diff
@@ -10868,4 +10891,167 @@ interface AVContentProposal : NSCopying
 		IntPtr Constructor (CMTime contentTimeForTransition, string title, [NullAllowed] UIImage previewImage);
 	}
 #endif
+
+	[NoWatch]
+	partial interface IAVContentKeySessionDelegate {}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3), NoWatch]
+	[Protocol, Model]
+	[BaseType (typeof (NSObject))]
+	interface AVContentKeySessionDelegate
+	{
+		[Abstract]
+		[Export ("contentKeySession:didProvideContentKeyRequest:")]
+		void DidProvideContentKeyRequest (AVContentKeySession session, AVContentKeyRequest keyRequest);
+
+		[Export ("contentKeySession:didProvideRenewingContentKeyRequest:")]
+		void DidProvideRenewingContentKeyRequest (AVContentKeySession session, AVContentKeyRequest keyRequest);
+
+		[Export ("contentKeySession:didProvidePersistableContentKeyRequest:")]
+		void DidProvidePersistableContentKeyRequest (AVContentKeySession session, AVPersistableContentKeyRequest keyRequest);
+
+		[Export ("contentKeySession:contentKeyRequest:didFailWithError:")]
+		void DidFail (AVContentKeySession session, AVContentKeyRequest keyRequest, NSError err);
+
+		[Export ("contentKeySession:shouldRetryContentKeyRequest:reason:")]
+		bool ShouldRetryContentKeyRequest (AVContentKeySession session, AVContentKeyRequest keyRequest, string retryReason);
+
+		[Export ("contentKeySessionContentProtectionSessionIdentifierDidChange:")]
+		void DidChange (AVContentKeySession session);
+	}
+
+	[NoWatch]
+	partial interface IAVContentKeyRecipient {}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3), NoWatch]
+	[Protocol]
+	interface AVContentKeyRecipient {
+		[Abstract]
+		[Export ("mayRequireContentKeysForMediaDataProcessing")]
+		bool MayRequireContentKeysForMediaDataProcessing { get; }
+	}
+
+	[TV (10,2), Mac (10,12,4), iOS (10,3), NoWatch]
+	[BaseType (typeof (NSObject))]
+	interface AVContentKeySession {
+
+		// used with Create (string keySystem, [NullAllowed] NSUrl storageUrl);
```

Comment:
interesting comment, let's have a `AVKeySystem` smart enum so we can avoid exposing the field.